packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -42,7 +42,9 @@ class _ProductImageViewerState extends State<ProductImageViewer> {
   @override
   void initState() {
     imageData = widget.imageData;
-    imageProvider = NetworkImage(imageData.imageUrl!);
+    imageProvider = NetworkImage(
+      imageData.imageUrl ?? '',
```

Comment:
That's not possible: if there's nothing to display you shouldn't be there with a non-existing image.

Reply:
@omkarChend1kar Thinking again about it, it's not that simple either, as the `imageUrl` can be refreshed from outside (perhaps not yet, but that's part of #3018 that I'm implementing).

That would mean that if on the gallery you click on an existing image, you go to a swipe page with all images, existing or non existing. On a non existing image, you have to display something specific like the "no picture" logo, with a "new image" button: that's what you should change from your PR code.
And if on the gallery you click directly on a non existing image, you go directly to the "new image" feature (like now).

Reply:
@monsieurtanuki Actually when imageUrl is without proper path or let say empty string, As I have declared there, It shows that No image icon, That's why I was putting empty string there instead of passing null, Since it was throwing nulll exception, After what you said about this, I thought of filtering list with non null Image paths and then show only uploaded images of whichever category would be displayed, 

But I guess this is what want right ,To show the icon/ text for non-existing images.

![noimagedisplay](https://user-images.githubusercontent.com/71365710/202195038-4e349db4-3756-4319-9f08-3e6eb5e98d7c.jpg)

Reply:
@omkarChend1kar That's correct - for the "no image" cases we need:
* a specific "no image" logo, instead of the image
* a specific "add" button label, instead of "edit"
* a specific "add" action, instead of "edit"